src/nxt_array.h

```diff
@@ -9,34 +10,32 @@
 
 
 typedef struct {
-    void      *elts;
-    /* nelts has uint32_t type because it is used most often. */
-    uint32_t  nelts;
-    uint16_t  size;
-    uint16_t  nalloc;
-    nxt_mp_t  *mem_pool;
```

Comment:
With this change, you made the array structure bigger without any reason.

Reply:
I'm starting from two comments:
```
/*
 * nxt_int_t corresponds to the most efficient integer type,
 * an architecture word. It is usually the long type
 * but on Win64 the long is int32_t, so pointer size suits better.
 * nxt_int_t must be no less than int32_t.
 */
```
```
/* nelts has uint32_t type because it is used most often. */
```
From this I conclude that `int32_t`, `uint32_t` are the most productive in the opinion of **Igor Sysoev** and my personal opinion. If we talk about the reality of using `uint16_t`, then I would like to see a place in the Unit program in the form of an array expanding to 65535 elements (`nalloc`). Therefore, I do not see a significant difference, except for the architectural one, which works faster with 32-bit values. Moreover, this type of structure is allocated once for each individual array. I think that the issue of frequency of calls is of course relevant in this case. I will review this structure to revert to the previous uint size for `nalloc` and `size`.

Reply:
`uint16_t` for `size` and `nalloc` has been chosen also by Igor, because it reduced the whole structure size by 8 bytes (on the 64-bits architectures). This is free economy as there are no such big arrays expected anywhere and the difference in performance for `uint16_t` in this case isn't measurable at all.